### PR TITLE
Feature/s7 coupon list

### PR DESCRIPTION
**Pull request by federicoboschini**
At the moment, the sample will show coupons only if real NearIT coupons are available for the profile. So, if the sample isn't linked to a valid NearIT app, every coupon list example will be empty.

### CODE REVIEW

File: app/src/main/java/com/nearit/ui_bindings_sample/CouponsActivity.java [hunk 1]
```diff
@@ -92,5 +92,94 @@ public void onClick(View view) {
...
+        });
+
+        Button validCouponList = (Button) findViewById(R.id.valid_coupon_list);
+        validCouponList.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                //  start an activity that shows a list of REAL valid coupons ONLY
+                startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
+                        .createCouponListIntentBuilder()
+                        .getValidOnly()
```
Comment (catt-stefano) [inline app/src/main/java/com/nearit/ui_bindings_sample/CouponsActivity.java]: Rename the builder get___ methods with onlyValidCoupons(), ...

File: docs/COUPON_LIST.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# NearIt-UI for coupon list
+
+Providing the list of the coupons earned by the user is a common feature of the apps that integrate NearIT.
```
Comment (catt-stefano) [inline docs/COUPON_LIST.md]: Providing a list of coupons earned by the user is a common feature of apps that integrate NearIT.

File: docs/COUPON_LIST.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+With NearIt-UI you can launch an activity or get a fragment that contains the coupons with our proposed sorting.
```
Comment (catt-stefano) [inline docs/COUPON_LIST.md]: .. that automatically fetches and displays coupons with ...

File: docs/COUPON_LIST.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+#### Introduction
+
+To understand the examples in this section it is important to know how NearIT coupons work.
+A coupon can have a date from which an user can redeem it and an expiration date: at a certain date a coupon could be in one of these states:
```
Comment (catt-stefano) [inline docs/COUPON_LIST.md]: remove the colons, they are used twice.

A coupon can have a date from which an user can redeem it and an expiration date.

File: docs/COUPON_LIST.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+- valid: redeemable, not expired and not already used.
+- not yet active: not yet redeemable
+- expired: current date is beyond the expiration one
+- redeemed: coupon code has been used
+
```
Comment (catt-stefano) [inline docs/COUPON_LIST.md]: Add:

With NearIT-UI, depending on its state, a coupon will appear in a different way.

File: docs/COUPON_LIST.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+```java
+startActivity(NearITUIBindings.getInstance(YourActivity.this)
+                        .createCouponListIntentBuilder()
+                        .getValidOnly()
+                        .build()
+                );
+```
+
+The same for the other states:
+- `getExpiredOnly()`
```
Comment (catt-stefano) [inline docs/COUPON_LIST.md]: rename those, see other comment

File: docs/COUPON_LIST.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+```java
+Fragment couponListFragment = NearITUIBindings.getInstance(this)
+                .createCouponListFragmentBuilder()
+                .build();
+```
+
+This builder provides the same methods of the previous, so you can show the list you prefer.
+
+Additionally, using both of the builders you are able to set the behaviour and look of the coupon detail that it is going to be shown when an user taps on one item of the list.
```
Comment (catt-stefano) [inline docs/COUPON_LIST.md]: both -> either

**Review by catt-stefano — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/nearit/ui_bindings_sample/CouponsActivity.java (hunk 1 above)
Comment (federicoboschini) [reply app/src/main/java/com/nearit/ui_bindings_sample/CouponsActivity.java]: done

File: docs/COUPON_LIST.md (hunk 2 above)
Comment (federicoboschini) [reply docs/COUPON_LIST.md]: done

File: docs/COUPON_LIST.md (hunk 3 above)
Comment (federicoboschini) [reply docs/COUPON_LIST.md]: done

File: docs/COUPON_LIST.md (hunk 4 above)
Comment (federicoboschini) [reply docs/COUPON_LIST.md]: done

File: docs/COUPON_LIST.md (hunk 5 above)
Comment (federicoboschini) [reply docs/COUPON_LIST.md]: done

File: docs/COUPON_LIST.md (hunk 6 above)
Comment (federicoboschini) [reply docs/COUPON_LIST.md]: done

File: docs/COUPON_LIST.md (hunk 7 above)
Comment (federicoboschini) [reply docs/COUPON_LIST.md]: done

**Pull request merged**